requirements-dev.txt

```diff
@@ -1,6 +1,10 @@
 -r requirements.txt
 black
+editorconfig-checker
 flake8
+isort
 lintrunner
 lintrunner-adapters
 pre-commit
+pylint
+ruff
```

Comment:
I suggest pinning the ruff version because there will be new rules w/ every new version and fail ci. Better let dependabot handle updates.

Reply:
Reference: https://github.com/onnx/onnx/blob/main/.github/dependabot.yml

Reply:
done